### PR TITLE
Feat: Added integration tests for new build image option

### PR DESCRIPTION
**Pull request by qingchm**
#### Which issue(s) does this change fix?
N.A.

#### Why is this change necessary?
To add integration tests for new option --build-image

#### How does it address the issue?
N.A.

#### What side effects does this change have?
N.A.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [X] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [X] Write/update integration tests
- [X] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: installer/pyinstaller/build-linux.sh [hunk 1]
```diff
@@ -2,6 +2,7 @@
 binary_zip_filename=$1
 python_library_zip_filename=$2
 python_version=$3
+nightly_build=$4
```
Comment (hoffa) [inline installer/pyinstaller/build-linux.sh:5]: Why is this here?

Comment (qingchm) [reply installer/pyinstaller/build-linux.sh:5]: The this due to when I pulled from the feature branch, my branch was already ahead of the feature branch. So it included extra commits. Not sure if any good solutions here. Will discuss about this as a part of the release improvements.
